tests/resources/helper_functions.py

```diff
@@ -350,3 +351,37 @@ def int_units(amount, num_decimals):
     unit_value = Decimal(10) ** num_decimals
 
     return int(decimal_amount * unit_value)
+
+
+@enforce_types
+def get_mock_provider_fees(mock_type, wallet, valid_until=0):
+    config = get_config_dict()
+    provider_fee_address = wallet.address
+    provider_fee_token = get_address_of_type(config, mock_type)
+    provider_fee_amount = 0
+    provider_data = json.dumps({"timeout": 0}, separators=(",", ":"))
+
+    message = Web3.solidityKeccak(
+        ["bytes", "address", "address", "uint256", "uint256"],
+        [
+            Web3.toHex(Web3.toBytes(text=provider_data)),
+            wallet.address,
+            provider_fee_token,
+            provider_fee_amount,
+            valid_until,
+        ],
+    )
+
+    signed = network.web3.eth.sign(wallet.address, data=message)
+    signature = split_signature(signed)
+
+    return {
+        "providerFeeAddress": provider_fee_address,
+        "providerFeeToken": provider_fee_token,
+        "providerFeeAmount": provider_fee_amount,
+        "v": signature.v,
+        "r": signature.r,
+        "s": signature.s,
+        "validUntil": valid_until,
+        "providerData": Web3.toHex(Web3.toBytes(text=provider_data)),
+    }
```

Comment:
Having a helper method for _tests_ to abstract away the messiness of messages & signing doesn't work for actual _users_. 

We need to properly abstract away this messiness for actual users.

Reply:
this is not it. We use a response from provider when we use provider fees in ocean.py. However, for tests, not to publish and call provider every time, we simulate a message from provider with the right data.

Reply:
the main issue with previous tests were that they were not simulating the response from provider, but instead doing all the signing and generating for every parameter. Now it simulates provider better and in just one place. It is just a test method, much like a fixture, and actual users don't need it.